File: 2095/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if not head.next:
            return head.next
        
        single_node_jump = head
        double_node_jump = head.next
        
        while double_node_jump.next and double_node_jump.next.next:
            single_node_jump = single_node_jump.next
            double_node_jump = double_node_jump.next.next
            
        else:
            single_node_jump.next = single_node_jump.next.next
        
        return head

File: 2095/solution.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode deleteMiddle(ListNode head) {
        
        if (head.next == null){
            return head.next;
        }
        
        ListNode single_jump = head;
        ListNode double_jump = head.next;
        
        while (double_jump.next != null && double_jump.next.next != null){
            single_jump = single_jump.next;
            double_jump = double_jump.next.next;
        }
        
        single_jump.next = single_jump.next.next;
        
        return head;
    }
}